src/main/java/seedu/toluist/controller/AddTaskController.java

```diff
@@ -54,8 +55,13 @@ public void execute(String command) {
 
         String taskPriority = tokens.get(TaskTokenizer.KEYWORD_TASK_PRIORITY);
 
+        String recurringFrequency = tokens.get(TaskTokenizer.KEYWORD_TASK_RECURRING_FREQUENCY);
+
+        String recurringUntilEndDateToken = tokens.get(TaskTokenizer.KEYWORD_TASK_RECURRING_UNTIL_END_DATE);
+        LocalDateTime recurringUntilEndDate = DateTimeUtil.parseDateString(recurringUntilEndDateToken);
+
         commandResult = add(todoList, description, eventStartDateTime, eventEndDateTime,
```

Comment:
sigh what to do im accepting all sorts of methods. The add and update function is even nastier with all the if-else logic